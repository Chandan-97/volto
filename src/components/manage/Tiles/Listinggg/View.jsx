/**
 * Listing tile.
 * @module components/manage/Tiles/Listing/View
 */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import { searchContent, resetSearchContent } from '../../../../actions';

@connect(
  state => ({
    searchSubrequests: state.search.subrequests,
  }),
  dispatch =>
    bindActionCreators(
      {
        searchContent,
        resetSearchContent,
      },
      dispatch,
    ),
)
/**
 * Listing tile class
 * @class View
 * @extends Component
 */
export default class View extends Component {
  /**
   * Property types.
   * @property {Object} propTypes Property types.
   * @static
   */
  static propTypes = {
    tile: PropTypes.string.isRequired,
    data: PropTypes.objectOf(PropTypes.any).isRequired,
    searchContent: PropTypes.func.isRequired,
    resetSearchContent: PropTypes.func.isRequired,
    searchSubrequests: PropTypes.objectOf(PropTypes.any),
  };

  static defaultProps = {
    searchSubrequests: {},
  };

  /**
   * Component did mount
   * @method componentDidMount
   * @returns {undefined}
   */
  componentDidMount() {
    const { query } = this.props.data;
    if (query) {
      // Use subrequests to fetch tile data
      this.props.searchContent('', query, this.props.tile);
    }
  }

  /**
   * Component will unmount. Reset loaded content.
   * @method componentWillUnmount
   * @returns {undefined}
   */
  componentWillUnmount() {
    this.props.resetSearchContent(this.props.tile);
  }

  render() {
    return <div className="tile listing" />;
  }
}
