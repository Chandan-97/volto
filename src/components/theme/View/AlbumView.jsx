/**
 * Album view component.
 * @module components/theme/View/AlbumView
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Helmet } from '@plone/volto/helpers';
import { Link } from 'react-router-dom';
import { Card, Container, Image, Icon } from 'semantic-ui-react';
import { FormattedMessage } from 'react-intl';
import { Button, Header, Modal } from 'semantic-ui-react';

/**
 * Album view component class.
 * @function AlbumView
 * @param {Object} content Content object.
 * @returns {string} Markup of the component.
 */

class AlbumView extends Component {
  constructor(props) {
    super(props);

    this.state = {
      openIndex: undefined,
    };

    this.nextImage = this.nextImage.bind(this);
    this.prevImage = this.prevImage.bind(this);
  }

  nextImage() {
    const openIndex =
      (this.state.openIndex + 1) % this.props.content.items.length;
    this.setState({
      openIndex,
    });
  }

  prevImage() {
    const openIndex =
      (this.state.openIndex - 1) % this.props.content.items.length;
    this.setState({
      openIndex,
    });
  }

  render() {
    const { content } = this.props;
    console.log(this.state);
    return (
      <Container className="view-wrapper">
        <Helmet title={content.title} />
        <article id="content">
          <header>
            <h1 className="documentFirstHeading">{content.title}</h1>
            {content.description && (
              <p className="documentDescription">{content.description}</p>
            )}
          </header>
          <section id="content-core">
            <Image.Group size="small">
              {content.items.map((item, index) => (
                <span key={item.url}>
                  {item.image && (
                    <Modal
                      open={this.state.openIndex === index}
                      trigger={
                        <Image
                          alt={
                            item.image_caption ? item.image_caption : item.title
                          }
                          src={item.image.scales.thumb.download}
                          onClick={() => {
                            this.setState({
                              openIndex: index,
                            });
                          }}
                        />
                      }
                    >
                      <Modal.Header>{item.title}</Modal.Header>
                      <button onClick={this.nextImage}>
                        <Icon name="left arrow" />
                      </button>
                      <Modal.Content image>
                        <Image
                          wrapped
                          size="large"
                          alt={
                            item.image_caption ? item.image_caption : item.title
                          }
                          src={item.image.scales.large.download}
                        />
                        <Modal.Description>
                          <p>{item.description}</p>
                        </Modal.Description>
                      </Modal.Content>
                      <button onClick={this.nextImage}>
                        <Icon name="right arrow" />
                      </button>
                    </Modal>
                  )}
                  <Link to={item.url} title={item['@type']}>
                    {item.title}
                  </Link>
                </span>
              ))}
            </Image.Group>
          </section>
        </article>
      </Container>
    );
  }
}

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
AlbumView.propTypes = {
  /**
   * Content of the object
   */
  content: PropTypes.shape({
    /**
     * Title of the object
     */
    title: PropTypes.string,
    /**
     * Description of the object
     */
    description: PropTypes.string,
    /**
     * Child items of the object
     */
    items: PropTypes.arrayOf(
      PropTypes.shape({
        /**
         * Title of the item
         */
        title: PropTypes.string,
        /**
         * Description of the item
         */
        description: PropTypes.string,
        /**
         * Url of the item
         */
        url: PropTypes.string,
        /**
         * Image of the item
         */
        image: PropTypes.object,
        /**
         * Image caption of the item
         */
        image_caption: PropTypes.string,
        /**
         * Type of the item
         */
        '@type': PropTypes.string,
      }),
    ),
  }).isRequired,
};

export default AlbumView;
