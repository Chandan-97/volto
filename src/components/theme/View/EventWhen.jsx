import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import cx from 'classnames';

const datesForDisplay = (start, end) => {
  const mStart = moment(start);
  const mEnd = moment(end);
  const sameDay = mStart.isSame(mEnd, 'day');
  const sameTime = mStart.isSame(mEnd, 'minute');
  return {
    sameDay,
    sameTime,
    startDate: mStart.format('ll'),
    startTime: mStart.format('LT'),
    endDate: mEnd.format('ll'),
    endTime: mEnd.format('LT'),
  };
};

const EventWhen = ({ start, end = '', whole_day, open_end }) => {
  const datesInfo = datesForDisplay(start, end);
  // TODO I18N INTL
  return (
    <p
      className={cx('event-when', {
        'same-day': datesInfo.sameDay,
        'same-time': datesInfo.sameTime,
        'whole-day': whole_day,
        'open-end': open_end,
      })}
    >
      {!datesInfo.sameDay ? (
        <>
          <span className="start">
            <span className="start-date">{datesInfo.startDate}</span>
            {!whole_day && (
              <>
                {/* Plone has an optional word based on locale here */}
                <span> </span>
                <span className="start-time">{datesInfo.startTime}</span>
              </>
            )}
          </span>
          &nbsp;to&nbsp;
          <span className="end">
            <span className="end-date">{datesInfo.endDate}</span>
            {!whole_day && (
              <>
                {/* Plone has an optional word based on locale here */}
                <span> </span>
                <span className="end-time">{datesInfo.endTime}</span>
              </>
            )}
          </span>
        </>
      ) : (
        <>
          {whole_day && (
            <span className="start-date">{datesInfo.startDate}</span>
          )}
          {open_end && !whole_day && (
            <>
              <span className="start-date">{datesInfo.startDate}</span>
              &nbsp;from&nbsp;
              <span className="start-time">{datesInfo.startTime}</span>
            </>
          )}
          {!(whole_day || open_end) && (
            <>
              <span className="start-date">{datesInfo.startDate}</span>
              &nbsp;from&nbsp;
              <span className="start-time">{datesInfo.startTime}</span>
              &nbsp;to&nbsp;
              <span className="end-time">{datesInfo.endTime}</span>
            </>
          )}
        </>
      )}
    </p>
  );
};

EventWhen.propTypes = {
  start: PropTypes.string.isRequired,
  end: PropTypes.string,
  whole_day: PropTypes.bool,
  open_end: PropTypes.bool,
};

export default EventWhen;
