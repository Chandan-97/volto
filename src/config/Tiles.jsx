import { defineMessages } from 'react-intl';

import ViewDescriptionTile from '@plone/volto/components/manage/Tiles/Description/View';
import ViewHeroImageLeftTile from '@plone/volto/components/manage/Tiles/HeroImageLeft/View';
import ViewHTMLTile from '@plone/volto/components/manage/Tiles/HTML/View';
import ViewImageTile from '@plone/volto/components/manage/Tiles/Image/View';
import ViewListingTile from '@plone/volto/components/manage/Tiles/Listing/View';
import ViewMapTile from '@plone/volto/components/manage/Tiles/Maps/View';
import ViewTextTile from '@plone/volto/components/manage/Tiles/Text/View';
import ViewTitleTile from '@plone/volto/components/manage/Tiles/Title/View';
import ViewVideoTile from '@plone/volto/components/manage/Tiles/Video/View';

import EditDescriptionTile from '@plone/volto/components/manage/Tiles/Description/Edit';
import EditHeroImageLeftTile from '@plone/volto/components/manage/Tiles/HeroImageLeft/Edit';
import EditHTMLTile from '@plone/volto/components/manage/Tiles/HTML/Edit';
import EditImageTile from '@plone/volto/components/manage/Tiles/Image/Edit';
import EditListingTile from '@plone/volto/components/manage/Tiles/Listing/Edit';
import EditMapTile from '@plone/volto/components/manage/Tiles/Maps/Edit';
import EditTextTile from '@plone/volto/components/manage/Tiles/Text/Edit';
import EditTitleTile from '@plone/volto/components/manage/Tiles/Title/Edit';
import EditVideoTile from '@plone/volto/components/manage/Tiles/Video/Edit';

import codeSVG from '@plone/volto/icons/code.svg';
import globeSVG from '@plone/volto/icons/globe.svg';
import heroSVG from '@plone/volto/icons/hero.svg';
import listBulletSVG from '@plone/volto/icons/list-bullet.svg';

const customTiles = [
  {
    title: 'hero',
    icon: heroSVG,
  },
  {
    title: 'listing',
    icon: listBulletSVG,
  },
  {
    title: 'maps',
    icon: globeSVG,
  },
  {
    title: 'html',
    icon: codeSVG,
  },
];

const messagesTiles = defineMessages({
  title: {
    id: 'title',
    defaultMessage: 'Title',
  },
  description: {
    id: 'description',
    defaultMessage: 'Description',
  },
  text: {
    id: 'text',
    defaultMessage: 'Text',
  },
  listing: {
    id: 'listing',
    defaultMessage: 'Listing',
  },
  image: {
    id: 'image',
    defaultMessage: 'Image',
  },
  video: {
    id: 'video',
    defaultMessage: 'Video',
  },
});

const defaultTilesViewMap = {
  title: ViewTitleTile,
  description: ViewDescriptionTile,
  text: ViewTextTile,
  listing: ViewListingTile,
  image: ViewImageTile,
  video: ViewVideoTile,
  hero: ViewHeroImageLeftTile,
  maps: ViewMapTile,
  html: ViewHTMLTile,
};

const defaultTilesEditMap = {
  title: EditTitleTile,
  description: EditDescriptionTile,
  text: EditTextTile,
  listing: EditListingTile,
  image: EditImageTile,
  video: EditVideoTile,
  hero: EditHeroImageLeftTile,
  maps: EditMapTile,
  html: EditHTMLTile,
};

export { customTiles, defaultTilesViewMap, defaultTilesEditMap, messagesTiles };
