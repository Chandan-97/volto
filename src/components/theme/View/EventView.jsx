/**
 * EventView view component.
 * @module components/theme/View/EventView
 */

import React from 'react';
import PropTypes from 'prop-types';
import Helmet from 'react-helmet';
import { Container, Image, Segment, Header } from 'semantic-ui-react';

import EventWhen from './EventWhen';

/**
 * EventView view component class.
 * @function EventView
 * @params {object} content Content object.
 * @returns {string} Markup of the component.
 */
const EventView = ({ content }) => (
  <Container className="view-wrapper event-view">
    <Helmet title={content.title} />
    {content.title && <h1 className="documentFirstHeading">{content.title}</h1>}
    {content.description && (
      <p className="documentDescription">{content.description}</p>
    )}
    {content.image && (
      <Image
        className="document-image"
        src={content.image.scales.thumb.download}
        floated="right"
      />
    )}
    <Segment floated="right">
      {/* TODO I18N INTL */}
      {content.subjects.length > 0 && (
        <>
          <Header dividing sub>
            What
          </Header>
          {content.subjects.map((subj, idx) => (
            <span key={idx} className="event-subject">
              {subj}
            </span>
          ))}
        </>
      )}
      <Header dividing sub>
        When
      </Header>
      <EventWhen
        start={content.start}
        end={content.end}
        whole_day={content.whole_day}
        open_end={content.open_end}
      />
      {content.location && (
        <>
          <Header dividing sub>
            Where
          </Header>
          <p>{content.location}</p>
        </>
      )}
      {content.contact_name && (
        <>
          <Header dividing sub>
            Contact Name
          </Header>
          <p>
            {content.contact_email ? (
              <a href={`mailto:${content.contact_email}`}>
                {content.contact_name}
              </a>
            ) : (
              content.contact_name
            )}
          </p>
        </>
      )}
      {content.contact_phone && (
        <>
          <Header dividing sub>
            Contact Phone
          </Header>
          <p>{content.contact_phone}</p>
        </>
      )}
      {content.attendees.length > 0 && (
        <>
          <Header dividing sub>
            Attendees
          </Header>
          {content.attendees.map((attendee, idx) => (
            <span key="idx" className="event-attendee">
              {attendee}
            </span>
          ))}
        </>
      )}
      {content.event_url && (
        <>
          <Header dividing sub>
            Web
          </Header>
          <p>
            <a href={content.event_url}>Visit external website</a>
          </p>
        </>
      )}
    </Segment>
    {content.text && (
      <div dangerouslySetInnerHTML={{ __html: content.text.data }} />
    )}
  </Container>
);

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
EventView.propTypes = {
  content: PropTypes.shape({
    title: PropTypes.string,
    description: PropTypes.string,
    text: PropTypes.shape({
      data: PropTypes.string,
    }),
  }).isRequired,
};

export default EventView;
