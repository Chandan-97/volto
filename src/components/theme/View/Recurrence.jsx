import React from 'react';
import PropTypes from 'prop-types';
import { List } from 'semantic-ui-react';
import moment from 'moment';
import { datesForDisplay } from './EventWhen';

const findRecurrence = props => {
  const reccDates = [];
  const arr = props.recurrence.split(';');
  const days = arr[1].substring(arr[1].indexOf('=') + 1, arr[1].length);
  const num = arr[arr.length - 1];
  const count = num.match(/[0-9]/g);
  const newDates = datesForDisplay(props.start, props.end);
  if (arr[0].includes('DAILY')) {
    let firstDate = moment(newDates.startDate);
    for (let i = 1; i <= parseInt(count); i++) {
      reccDates.push(
        firstDate
          .add(1, 'days')
          ._d.toString()
          .substring(0, 15)
          .concat(` from ${newDates.startTime} to ${newDates.endTime}`),
      );
    }
  } else if (arr[0].includes('WEEKLY')) {
    let firstDate = moment(newDates.startDate);
    //console.log(days)
    days.split(',').map(item => {
      reccDates.push(
        firstDate
          .day(item)
          ._d.toString()
          .substring(0, 15)
          .concat(` from ${newDates.startTime} to ${newDates.endTime}`),
      );
    });
  } else if (arr[0].includes('MONTHLY')) {
    let firstDate = moment(newDates.startDate);
    //console.log(days, firstDate)
    for (let i = 1; i <= parseInt(count); i++) {
      reccDates.push(
        firstDate
          .day(days)
          ._d.toString()
          .substring(0, 15)
          .concat(` from ${newDates.startTime} to ${newDates.endTime}`),
      );
    }
  }
  return reccDates;
};

const Recurrence = props => {
  const reccDates = findRecurrence(props);
  if (reccDates.length > 0) {
    return (
      <div>
        <List items={reccDates} />
      </div>
    );
  }
  return (
    <>
      <List items={[props.recurrence]} />
    </>
  );
};

Recurrence.propTypes = {
  start: PropTypes.string.isRequired,
  end: PropTypes.string,
  recurrence: PropTypes.string.isRequired,
};

export default Recurrence;
