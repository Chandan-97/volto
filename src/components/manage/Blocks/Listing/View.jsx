/**
 * View listing block.
 * @module components/manage/Blocks/Listing/View
 */

import { get } from 'lodash';
import { Link } from 'react-router-dom';
import { List, Image } from 'semantic-ui-react';
import { settings } from '~/config';
import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import { searchContent, resetSearchContent } from '../../../../actions';

@connect(
  (state, props) => ({
    items: get(state.search.subrequests, [props.id, 'items'], []),
  }),
  dispatch =>
    bindActionCreators(
      {
        searchContent,
        resetSearchContent,
      },
      dispatch,
    ),
)
/**
 * Listing view component.
 * @class View
 * @extends Component
 */
export default class View extends Component {
  /**
   * Property types.
   * @property {Object} propTypes Property types.
   * @static
   */
  static propTypes = {
    id: PropTypes.string.isRequired,
    data: PropTypes.objectOf(PropTypes.any).isRequired,
    searchContent: PropTypes.func.isRequired,
    resetSearchContent: PropTypes.func.isRequired,
    items: PropTypes.arrayOf(PropTypes.any).isRequired,
    properties: PropTypes.objectOf(PropTypes.any).isRequired,
  };

  /**
   * Component did mount
   * @method componentDidMount
   * @returns {undefined}
   */
  componentDidMount() {
    const { query } = this.props.data;
    if (query) {
      // Use subrequests to fetch block data
      try {
        const options = {
          ...JSON.parse(query),
          fullobjects: 1,
        };
        this.props.searchContent('', options, this.props.id);
      } catch (err) {
        console.error('Listing Block View - componentDidMount: ', err.message);
      }
    }
  }

  /**
   * Component will unmount. Reset loaded content.
   * @method componentWillUnmount
   * @returns {undefined}
   */
  componentWillUnmount() {
    this.props.resetSearchContent(this.props.id);
  }

  /**
   * Render method.
   * @method render
   * @returns {string} Markup for the component.
   */
  render() {
    const { data, items, properties } = this.props;
    const folderItems = properties ? properties.items : [];
    const listingItems = data.query ? items : folderItems;

    if (listingItems.length === 0) {
      return <List className="block listing" />;
    }

    return (
      <List className="block listing">
        {listingItems.map(item => {
          const image = get(item, 'image.scales.mini.download', '').replace(
            settings.apiPath,
            '',
          );
          const url = item['@id'].replace(settings.apiPath, '');
          return (
            <List.Item key={item.UID}>
              {image && <Image avatar src={image} alt={item.title} />}
              <List.Content>
                <List.Header>
                  <Link to={url}>{item.title}</Link>
                </List.Header>
                <List.Description>{item.description}</List.Description>
              </List.Content>
            </List.Item>
          );
        })}
      </List>
    );
  }
}
