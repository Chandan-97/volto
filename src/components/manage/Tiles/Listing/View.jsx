/**
 * View listing tile.
 * @module components/manage/Tiles/Listing/View
 */

import { get } from 'lodash';
import { Link } from 'react-router-dom';
import { List, Image } from 'semantic-ui-react';
import { settings } from '~/config';
import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { FormattedMessage } from 'react-intl';

import { searchContent, resetSearchContent } from '../../../../actions';

@connect(
  (state, props) => ({
    items: get(state.search.subrequests, [props.tile, 'items'], []),
  }),
  dispatch =>
    bindActionCreators(
      {
        searchContent,
        resetSearchContent,
      },
      dispatch,
    ),
)
/**
 * Listing view component.
 * @class View
 * @extends Component
 */
export default class View extends Component {
  /**
   * Property types.
   * @property {Object} propTypes Property types.
   * @static
   */
  static propTypes = {
    tile: PropTypes.string.isRequired,
    data: PropTypes.objectOf(PropTypes.any).isRequired,
    searchContent: PropTypes.func.isRequired,
    resetSearchContent: PropTypes.func.isRequired,
    items: PropTypes.arrayOf(PropTypes.any).isRequired,
  };

  /**
   * Component did mount
   * @method componentDidMount
   * @returns {undefined}
   */
  componentDidMount() {
    const { query = '' } = this.props.data;
    if (query !== '') {
      // Use subrequests to fetch tile data
      const options = {
        ...JSON.parse(query),
        fullobjects: 1,
      };
      this.props.searchContent('', options, this.props.tile);
    }
  }

  /**
   * Component will unmount. Reset loaded content.
   * @method componentWillUnmount
   * @returns {undefined}
   */
  componentWillUnmount() {
    this.props.resetSearchContent(this.props.tile);
  }

  /**
   * Render method.
   * @method render
   * @returns {string} Markup for the component.
   */
  render() {
    const { data, items, properties } = this.props;
    debugger;
    const folderItems = properties ? properties.items : [];
    const listingItems = data.query !== '' ? items : folderItems;

    if (listingItems.length === 0) {
      return <List className="tile listing" />;
    }

    return (
      <List className="tile listing">
        {listingItems.map(item => {
          const image = get(item, 'image.scales.mini.download', '').replace(
            settings.apiPath,
            '',
          );
          const url = item['@id'].replace(settings.apiPath, '');
          return (
            <List.Item>
              {image && <Image avatar src={image} alt={item.title} />}
              <List.Content>
                <List.Header as="a">
                  <Link to={url}>{item.title}</Link>
                </List.Header>
                <List.Description>{item.description}</List.Description>
              </List.Content>
            </List.Item>
          );
        })}
      </List>
    );
  }
}
