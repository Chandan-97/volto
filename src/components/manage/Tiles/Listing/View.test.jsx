import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import { settings } from '~/config';

import { View } from './View';

const props = {
  data: {
    selectedItem: '/test',
  },
  getContent: () => {},
  resetContent: () => {},
  tile: 'test',
  contentSubrequests: {
    test: {
      data: {
        '@id': `${settings.apiUrl}/test`,
        image: {
          scales: {
            mini: {
              download: `${settings.apiUrl}/test/image.jpeg`,
            },
          },
        },
        title: 'Test',
        description: 'Summary',
      },
    },
  },
};

test('renders a summary box view component', () => {
  const component = renderer.create(
    <MemoryRouter>
      <View {...props} />
    </MemoryRouter>,
  );
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});
