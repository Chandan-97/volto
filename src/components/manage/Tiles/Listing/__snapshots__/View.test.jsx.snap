// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a summary box view component 1`] = `
<div
  className="ui items summary-box-item"
>
  <div
    className="item"
  >
    <a
      className="ui small image"
      href="/test"
      onClick={[Function]}
    >
      <img
        alt="Test"
        className="ui image"
        src="undefined/test/image.jpeg"
      />
    </a>
    <div
      className="content"
    >
      <div
        className="header"
      >
        <a
          href="/test"
          onClick={[Function]}
        >
          Test
        </a>
      </div>
      <div
        className="description"
      >
        Summary
      </div>
    </div>
  </div>
</div>
`;
